From 5b437bc9a8b53132bed6cab981ee3ae876f864e0 Mon Sep 17 00:00:00 2001
From: Sylvain Munaut <tnt@246tNt.com>
Date: Wed, 26 Aug 2015 07:28:57 -0400
Subject: [PATCH 06/17] grc/blocks: Fix XML RPC server to allow proper
 flowgraph termination

Without this, the XML RPC server thread will stay alive and the program
will never terminate.

By making it a 'daemon' thread, if it's the only thread left alive, python
will automatically terminate it.

Signed-off-by: Sylvain Munaut <tnt@246tNt.com>
---
 grc/blocks/xmlrpc_server.xml | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/grc/blocks/xmlrpc_server.xml b/grc/blocks/xmlrpc_server.xml
index 6c31bd1..602d444 100644
--- a/grc/blocks/xmlrpc_server.xml
+++ b/grc/blocks/xmlrpc_server.xml
@@ -11,7 +11,9 @@
 	<import>import threading</import>
 	<make>SimpleXMLRPCServer.SimpleXMLRPCServer(($addr, $port), allow_none=True)
 self.$(id).register_instance(self)
-threading.Thread(target=self.$(id).serve_forever).start()</make>
+self.$(id)_thread = threading.Thread(target=self.$(id).serve_forever)
+self.$(id)_thread.daemon = True
+self.$(id)_thread.start()</make>
 	<param>
 		<name>Address</name>
 		<key>addr</key>
-- 
2.1.4

